certificate_loader.go

```diff
@@ -55,6 +56,10 @@ func NewCertificateLoader() CertificateLoader {
 }
 
 func (c *CertificateLoader) Load(path string, password string) error {
+	if unix.Access(path, unix.W_OK) != nil {
+		return nil
+	}
```

Comment:
Were you seeing an error without this? Just trying to understand why it's required. 

Also, typically, with Go the pattern you see is that you would just try the operation and then handle the error, rather than checking in advance. I believe that is to avoid potential time-of-check, time-of-use (TOCTOU) race conditions. I don't 100% know that's an issue here, but we do try to follow that pattern in buildpacks. Would it be possible to do that here?

Reply:
I guess this was moved here b/c of DRY and to cover both cases, JKS and PKCS12. I'll try to clarify that.

Reply:
> Just trying to understand why it's required.

It isn't and we shouldn't probably have touched this. But `Load` will detect and load the certificates just to call `Write` which used to check the permission before. so the idea was to not do anything if we cannot write them anyway. The "problem" is that the `Write` will not raise an error. I think that used to be important for readonly rootfs, but that should work in the meantime.

But I guess we can just move it back since it is not related to this pr anyway.

Reply:
Ah, I see. Let's keep it in the original location for now. If there is a problem like you mentioned, open a new PR for it and we can move things around there. It'll just keep this PR from having a potential side effect.